### PR TITLE
Prevent circular references in the /Pages tree

### PR DESCRIPTION
**Pull request by Snuffleupagus**

### CODE REVIEW

**Comment by Rob--W** [comment]
If I'm not mistaken, this looks like something that could reasonably easily be covered by a unit test. Could you update the PR with a unit test?

**Comment by Snuffleupagus** [comment]
> Could you update the PR with a unit test?

While I already have a test-case locally, please note that I'd prefer to wait for @brendandahl to sign-off on that though.

File: web/pdf_rendering_queue.js [hunk 1]
```diff
@@ -164,9 +164,14 @@ class PDFRenderingQueue {
...
-        view.draw().finally(() => {
-          this.renderHighestPriority();
-        });
+        view
+          .draw()
+          .finally(() => {
+            this.renderHighestPriority();
+          })
+          .catch(() => {
+            /* Avoid "Uncaught promise" messages in the console. */
```
Comment (brendandahl) [inline web/pdf_rendering_queue.js]: Can we either log the error or remove the catch? Empty catch statements can make debugging very painful.

**Review by brendandahl — CHANGES_REQUESTED** [review]
Almost there, I'd also like to see a test case and the one comment addressed.

File: web/pdf_rendering_queue.js (hunk 1 above)
Comment (Snuffleupagus) [reply web/pdf_rendering_queue.js]: Fair points, any errors here are now logged (such that there's no "Uncaught promise" warnings) and a unit-test were added.

**Review by brendandahl — APPROVED** [review]

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Thank you!

**Pull request merged**